### ISSUE TITLE
PublishDir warning - Failed to publish file - published regardless.

### ISSUE THREAD

**Issue by paulderaadt** [nextflow-io/nextflow#1616]
## Bug report 
As a follow-up to https://groups.google.com/d/msg/nextflow/ULyv_tDJ0Pg/zJgKtbDWCgAJ

### Expected behavior and actual behavior

When publishing a file that has been sent to two channels in its process'  output, nextflow throws a warning, despite succesfully publishing the file. A `diff` on the source and target files shows succesful copying. It is not a fatal bug, but certainly it doesn't seem unreasonable to be able to output a file into multiple channels.

### Steps to reproduce the problem
input files:
```
$:ls files/
SampleA_L001_R1.fastq.gz  SampleA_L002_R1.fastq.gz  Sampleb_L001_R1.fastq.gz  Sampleb_L002_R1.fastq.gz

```

This code reproduces  the problem:
```
Channel
.fromPath('files/Sample*_R1.fastq.gz').
map{    Path path -> [path.getBaseName().toString().split("_")[0], path]}
.groupTuple()
.set { fastqs_1}


process a {
    publishDir "output", mode: "copy", pattern: "SampleA*"
    publishDir "output", mode: "copy", pattern: "Sampleb*"
    input:
    set pairid, file(reads) from fastqs_1.view()
    output:
    set pairid, file("${pairid}*.txt") into foo
    set pairid, file("${pairid}*.txt") into bar

    echo true
    script:
    """
    for i in {1..1000}
    do
        echo $pairid $reads >> ${pairid}1.txt
        echo $pairid $reads >> ${pairid}2.txt
    done
    """
}
foo.view()
bar.view()

```
### Program output 
```
nextflow main.nf -ansi-log false
N E X T F L O W  ~  version 20.04.1
Launching `main.nf` [elated_majorana] - revision: 0d8cc31c04
[Sampleb, [/mnt/nfs/testfolder/paul/nextflow_tests/files/Sampleb_L001_R1.fastq.gz, /mnt/nfs/testfolder/paul/nextflow_tests/files/Sampleb_L002_R1.fastq.gz]]
[SampleA, [/mnt/nfs/testfolder/paul/nextflow_tests/files/SampleA_L001_R1.fastq.gz, /mnt/nfs/testfolder/paul/nextflow_tests/files/SampleA_L002_R1.fastq.gz]]
[a2/7b7865] Submitted process > a (1)
[68/141880] Submitted process > a (2)
[SampleA, [/mnt/nfs/testfolder/paul/nextflow_tests/work/68/141880f97f61a04b3108b5d14fddb2/SampleA1.txt, /mnt/nfs/testfolder/paul/nextflow_tests/work/68/141880f97f61a04b3108b5d14fddb2/SampleA2.txt]]
[SampleA, [/mnt/nfs/testfolder/paul/nextflow_tests/work/68/141880f97f61a04b3108b5d14fddb2/SampleA1.txt, /mnt/nfs/testfolder/paul/nextflow_tests/work/68/141880f97f61a04b3108b5d14fddb2/SampleA2.txt]]
WARN: Failed to publish file: /mnt/nfs/testfolder/paul/nextflow_tests/work/68/141880f97f61a04b3108b5d14fddb2/SampleA1.txt; to: /mnt/nfs/testfolder/paul/nextflow_tests/output/SampleA1.txt [copy] -- See log file for details
WARN: Failed to publish file: /mnt/nfs/testfolder/paul/nextflow_tests/work/68/141880f97f61a04b3108b5d14fddb2/SampleA2.txt; to: /mnt/nfs/testfolder/paul/nextflow_tests/output/SampleA2.txt [copy] -- See log file for details
[Sampleb, [/mnt/nfs/testfolder/paul/nextflow_tests/work/a2/7b786572082668ae6271f9339fccb3/Sampleb1.txt, /mnt/nfs/testfolder/paul/nextflow_tests/work/a2/7b786572082668ae6271f9339fccb3/Sampleb2.txt]]
[Sampleb, [/mnt/nfs/testfolder/paul/nextflow_tests/work/a2/7b786572082668ae6271f9339fccb3/Sampleb1.txt, /mnt/nfs/testfolder/paul/nextflow_tests/work/a2/7b786572082668ae6271f9339fccb3/Sampleb2.txt]]
WARN: Failed to publish file: /mnt/nfs/testfolder/paul/nextflow_tests/work/a2/7b786572082668ae6271f9339fccb3/Sampleb2.txt; to: /mnt/nfs/testfolder/paul/nextflow_tests/output/Sampleb2.txt [copy] -- See log file for details
WARN: Failed to publish file: /mnt/nfs/testfolder/paul/nextflow_tests/work/a2/7b786572082668ae6271f9339fccb3/Sampleb1.txt; to: /mnt/nfs/testfolder/paul/nextflow_tests/output/Sampleb1.txt [copy] -- See log file for details
```
Relevant log file section:
```
Jun-02 10:09:16.636 [FileTransfer-thread-3] WARN  nextflow.processor.PublishDir - Failed to publish file: /mnt/nfs/testfolder/paul/nextflow_tests/work/68/141880f97f61a04b3108b5d14fddb2/SampleA1.txt; to: /mnt/nfs/testfolder/paul/nextflow_tests/output/SampleA1.txt [copy] -- See log file for details
java.nio.file.NoSuchFileException: /mnt/nfs/testfolder/paul/nextflow_tests/output/SampleA1.txt
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.implDelete(UnixFileSystemProvider.java:244)
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
	at java.nio.file.Files.delete(Files.java:1126)
	at nextflow.file.FileHelper.deletePath(FileHelper.groovy:901)
	at nextflow.processor.PublishDir.processFile(PublishDir.groovy:279)
	at nextflow.processor.PublishDir.safeProcessFile(PublishDir.groovy:259)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:101)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1217)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1041)
	at org.codehaus.groovy.runtime.InvokerHelper.invokePogoMethod(InvokerHelper.java:1011)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:994)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethodSafe(InvokerHelper.java:97)
	at nextflow.processor.PublishDir$_apply1_closure1.doCall(PublishDir.groovy:232)
	at nextflow.processor.PublishDir$_apply1_closure1.call(PublishDir.groovy)
	at groovy.lang.Closure.run(Closure.java:486)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```
### Environment 

* Nextflow version: version 20.04.1
* Java version: 
openjdk version "1.8.0_242"
OpenJDK Runtime Environment (build 1.8.0_242-b08)
OpenJDK 64-Bit Server VM (build 25.242-b08, mixed mode)
* Operating system: Linux
* Bash version: GNU bash, version 4.2.46(2)-release (x86_64-redhat-linux-gnu)

### Additional context

nextflow version 19.04.1 runs the same code without any warnings thrown.

**Comment by pditommaso** [comment]
This may be fixed by 5f3b69c1 include in release `20.05.0-edge`. Please give it a try.

**Comment by paulderaadt** [comment]
> This may be fixed by [5f3b69c](https://github.com/nextflow-io/nextflow/commit/5f3b69c19e70e9245aba25ca668d7905eea09cb6) included in release `20.05.0-edge`. Please give it a try.

Thanks @pditommaso , version 20.05.0-edge indeed runs the code above without warnings.

**Comment by pditommaso** [comment]
Cool
